.gitlab-ci.yml

```diff
@@ -43,41 +42,70 @@ build-for-pypi:
     paths:
       - dist/*
 
-delete-vm:
+build-msi:
   rules:
-    - if: '$CI_PIPELINE_SOURCE == "push"'  #run this job only on push event
+    - if: '$CI_PIPELINE_SOURCE == "push"'
+    - if: '$CI_PIPELINE_SOURCE == "schedule"'
+    - if: '$CI_PIPELINE_SOURCE == "web"'
   tags:
-    - shell
-  stage: cleanup
+    - dinvm
+  stage: build
   script:
-    - vboxmanage unregistervm ubuntu-runner-runner-hszxezQU-concurrent-0 --delete
+    - make wine-build
+  artifacts:
+    paths:
+      - wine-build/*.exe
+      - wine-build/*.msi
 
-publish-pypi:
-  when: manual   #run this job only on manual user interaction in the webinterface
-  allow_failure: false
-  only:
-    - pushes
+upload-nightly:
+  stage: deploy
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "schedule"'
+  tags:
+    - uind
   image: ubuntu:20.04
+  before_script:
+    - apt-get update -y
+    - apt-get install -y git
+    - apt-get -y install zip
+  script:
+    - chmod +x nightly_upload.sh
+    - mkdir nightly-build
+    - mv wine-build/*.msi nightly-build
+    - mv wine-build/*.exe nightly-build
+    - zip -r pynitrokey.zip nightly-build
+    - ./nightly_upload.sh pynitrokey.zip
+
+publish-pypi:
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "web"'
   tags:
     - dinvm
   stage: deploy
   script:
-    - make update-venv
+    - make CI
     - echo "[distutils]" >> ~/.pypirc
     - echo "index-servers =" >> ~/.pypirc
-    - echo "    test-pypi" >> ~/.pypirc
+    - echo "    pypi" >> ~/.pypirc
     - echo "" >> ~/.pypirc
-    - echo "[test-pypi]" >> ~/.pypirc
+    - echo "[pypi]" >> ~/.pypirc
     - echo "repository = https://upload.pypi.org/legacy/" >> ~/.pypirc
     - echo "username = __token__" >> ~/.pypirc
-    - echo "${PYPI_TOKEN}" >> ~/.pypirc
+    - echo "password=${PYPI_TOKEN}" >> ~/.pypirc
     - make publish
 
-delete-publish-vm:
+publish-msi:
   rules:
-    - if: '$CI_PIPELINE_SOURCE == "push"'  #run this job only on push event
+    - if: '$CI_PIPELINE_SOURCE == "web"'
+  image: ubuntu:20.04
   tags:
-    - shell
-  stage: deploy-cleanup
+    - uind
+  stage: deploy
   script:
-    - vboxmanage unregistervm ubuntu-runner-runner-hszxezQU-concurrent-0 --delete
\ No newline at end of file
+    - apt-get update -y
```

Comment:
Is `update` really needed? Perhaps this takes time unnecessarily.
How old is the Ubuntu image they provide?

Reply:
currently 14 days 
i will remove it from service jobs like pull but keep it in for all jobs that build stuff